### PR TITLE
[SPARK-32524][SQL][TESTS] CachedBatchSerializerSuite should clean up InMemoryRelation.ser

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to clean up `InMemoryRelation.ser` in `CachedBatchSerializerSuite`.

### Why are the changes needed?

SPARK-32274 makes SQL cache serialization pluggable.
```
[SPARK-32274][SQL] Make SQL cache serialization pluggable
```

This causes UT failures.
```
$ build/sbt "sql/testOnly *.CachedBatchSerializerSuite *.CachedTableSuite"
...
[info]   Cause: java.lang.IllegalStateException: This does not work. This is only for testing
[info]   at org.apache.spark.sql.execution.columnar.TestSingleIntColumnarCachedBatchSerializer.convertInternalRowToCachedBatch(CachedBatchSerializerSuite.scala:49)
...
[info] *** 30 TESTS FAILED ***
[error] Failed: Total 51, Failed 30, Errors 0, Passed 21
[error] Failed tests:
[error] 	org.apache.spark.sql.CachedTableSuite
[error] (sql/test:testOnly) sbt.TestsFailedException: Tests unsuccessful
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually.
```
$ build/sbt "sql/testOnly *.CachedBatchSerializerSuite *.CachedTableSuite"
[info] Tests: succeeded 51, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 51, Failed 0, Errors 0, Passed 51
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @cloud-fan,  @HyukjinKwon , @revans2 , @tgravescs 
This is the second approach according to the review comments.

**Review by HyukjinKwon — APPROVED** [review]
LGTM. I prefer this way but don't feel strongly if other people prefer the other way.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/CachedBatchSerializerSuite.scala [hunk 1]
```diff
@@ -120,6 +121,11 @@ class CachedBatchSerializerSuite  extends QueryTest with SharedSparkSession {
       classOf[TestSingleIntColumnarCachedBatchSerializer].getName)
   }
 
+  protected override def afterAll(): Unit = {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/CachedBatchSerializerSuite.scala:124]: let's do it in `beforeAll` as well, to guarantee that we are testing a fresh serializer in this test suite.

**Review by cloud-fan — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Thank you for reiview and approval, @HyukjinKwon .

File: sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/CachedBatchSerializerSuite.scala (hunk 1 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/CachedBatchSerializerSuite.scala:124]: Sure!

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/CachedBatchSerializerSuite.scala:124]: Oh. It should be here. `beforeAll` will leave the leftover and will break the other suite.

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/CachedBatchSerializerSuite.scala:124]: Do you want to have both beforeAll and afterAll?

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/CachedBatchSerializerSuite.scala:124]: Got it~ I missed `as well` in your comment.

**Comment by dongjoon-hyun** [comment]
Thanks, @cloud-fan . `beforeAll` is also added.

**Comment by cloud-fan** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
I am going to merge this since GitHub Actions builds passed.

**Comment by dongjoon-hyun** [comment]
Thank you, @cloud-fan and @HyukjinKwon .

**Comment by revans2** [comment]
Thanks for finding and fixing this for me.

**Comment by dongjoon-hyun** [comment]
Thanks, @revans2 ! :)

**Pull request closed without merging**